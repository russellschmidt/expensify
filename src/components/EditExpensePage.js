import React from 'react'
import {connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import {editExpense, removeExpense} from '../actions/expenses'

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  onRemove = (expense) => {
    this.props.removeExpense(this.props.expense)
    this.props.history.push('/')
  }
  render() {
    return(
      <div>
        <h1>Edit Expense</h1>
        <ExpenseForm 
          expense={this.props.expense} 
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onRemove}>
          Remove
        </button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
)

const mapDispatchToProps = (dispatch, props) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (expense) => dispatch(removeExpense(expense))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm 
//         expense={props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense))
//           props.history.push('/')
//         }}
//       />

//       <button onClick={() => {
//         props.dispatch(removeExpense(props.expense))
//         props.history.push('/')
//       }}>Remove</button>
//     </div>
//   )
// }

// const MapStateToProps = (state, props) => {
//   return {
//     expense: state.expenses.find((expense) => expense.id === props.match.params.id)
//   } 
// }


// export default connect(MapStateToProps)(EditExpensePage)