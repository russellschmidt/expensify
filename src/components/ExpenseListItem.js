import React from 'react'
import {Link} from 'react-router-dom'

export const ExpenseListItem = (props) => (
  <tr>
    <td><Link to={`/edit/${props.id}`}>{props.description}</Link></td>
    <td>${(props.amount / 100).toFixed(2)}</td>
    <td>{props.createdAt}</td>
  </tr>
)

export default ExpenseListItem