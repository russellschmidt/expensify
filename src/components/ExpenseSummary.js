import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'

import selectExpenses from '../selectors/expenses'
import getExpenseTotal from '../selectors/expenses-total'

export const ExpenseSummaryPage = (props) => {
  const formattedExpense = numeral(getExpenseTotal(props.expenses)).format('$0, 0.00')
  const expenseCount = props.expenses.length
  return (
    <div>
      <h3>Viewing {expenseCount} expense{ expenseCount === 1 ? '' : 's'} totaling {formattedExpense} </h3> 
    </div>
    )
  }

const mapStateToProps = (state, props) => ({
  expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseSummaryPage)
