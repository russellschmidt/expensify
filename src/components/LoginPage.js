import React from 'react'

const LoginPage = () => (
  <div>
    <button>Log in</button>
  </div>
)

export { LoginPage }