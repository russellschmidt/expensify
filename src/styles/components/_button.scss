.box-layout__button {
  background: $deep-blue;
  border: none;
  border-radius: 4px;
  box-shadow: 2px 2px 3px $shadow-grey;
  color: white;
  font-size: $font-size-large;
  font-weight: 300;
  height: 5rem;
  width: 20rem;
}