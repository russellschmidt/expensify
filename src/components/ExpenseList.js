import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    <table>
      <thead>
        <tr>
          <th>Description</th>
          <th>Amount</th>
          <th>Created At</th>
        </tr>
      </thead>
      <tbody>
        {props.expenses.map(expense => <ExpenseListItem {...expense} key={expense.id}/>)}
      </tbody>
    </table>
  </div>
) 

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

// ConnectedExpenseList
export default connect(mapStateToProps)(ExpenseList)