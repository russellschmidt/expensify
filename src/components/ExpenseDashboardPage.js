import React from 'react'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpenseSummary'
import ExpenseList from './ExpenseList'

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseSummary />
    <ExpenseList />
  </div>
)

export default ExpenseDashboardPage