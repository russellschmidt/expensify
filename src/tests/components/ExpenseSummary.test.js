import React from 'react'
import { shallow } from 'enzyme'
import {ExpenseSummaryPage} from '../../components/ExpenseSummary'

const expenses = [1,2,3]

test('should correctly render ExpensesSummaryPage with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummaryPage expenses={[1]}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSummaryPage with 3 expenses', () => {
  const wrapper = shallow(<ExpenseSummaryPage expenses={expenses}/>)
  expect(wrapper).toMatchSnapshot()
})