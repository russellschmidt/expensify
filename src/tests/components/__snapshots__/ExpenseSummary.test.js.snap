// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummaryPage with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     expense
     totaling 
    $0.00
     
  </h3>
</div>
`;

exports[`should correctly render ExpensesSummaryPage with 3 expenses 1`] = `
<div>
  <h3>
    Viewing 
    3
     expense
    s
     totaling 
    $0.00
     
  </h3>
</div>
`;
